### DISCUSSION TITLE
Writing a mesh reader for an OpenFOAM mesh.

### DISCUSSION THREAD

**Discussion by danielwardstfc** [idaholab/moose#26274] [Q&A General]
Hi, I'm looking to write a mesh reader to take a mesh in an OpenFOAM directory and load it into MOOSE.  I've tried to dig into the source a bit and have got to moose/framework/include/parser/Parser.h and found a message asking me to contact the developers.

I'm trying to find out how mesh files are currently parsed and read in and I was wondering whether you could point me in the right direction.

Thanks,

Daniel

**Comment by GiudGiud** [comment]
Hello

All the mesh readers in MOOSE are coming from libmesh. There's even a few in libmesh we have not created interfaces for in MOOSE.

Once you have the mesh reader merged on the libmesh side, we ll want to simply make the FileMeshGenerator in MOOSE use it.

Best,
Guillaume
